frontend/public/components/build-pipeline.tsx

```diff
@@ -0,0 +1,174 @@
+  const json = _.attempt(JSON.parse, status);
+  return _.isError(json) ? {} : json;
+};
+const getJenkinsLogURL = (resource: K8sResourceKind): string => _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-log-url']);
+const getJenkinsBuildURL = (resource: K8sResourceKind): string => _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-build-uri']);
+
+const BuildSummaryStatusIcon: React.SFC<BuildSummaryStatusIconProps> = ({status}) => {
+  const statusClass = _.lowerCase(status);
+  const icon = ({
+    new: 'fa-hourglass-o',
+    pending: 'fa-hourglass-half',
+    running: 'fa-refresh fa-spin',
+    complete: 'fa-check-circle',
+    failed: 'fa-times-circle'
+  })[statusClass];
+  const hide = icon ? 'hide' : '';
+  return <React.Fragment>
+    <span className={`build-pipeline__status-icon build-pipeline__status-icon--${statusClass}`}>
+      <span aria-hidden="true">
+        <span className={`fa ${icon} fa-fw`}></span>
+      </span>
+    </span>
+
+    <span className={`build-pipeline__status-icon ${hide}`}>
+      <span className="fa fa-refresh fa-spin" aria-hidden="true"></span>
+    </span>
+  </React.Fragment>;
+};
+
+const BuildLogLink: React.SFC<BuildLogLinkProps> = ({obj}) => {
+  const link = getJenkinsLogURL(obj);
+  return link ? <div className="build-pipeline__link">
+    <a href={link} className="build-pipeline__log-link" target="_blank" rel="noopener noreferrer">View Log</a>
+  </div> : null;
+};
+
+const StagesNotStarted: React.SFC<StagesNotStartedProps> = ({stages}) => {
+  return !stages.length ? <div className="build-pipeline__stage build-pipeline__stage--none">No stages have started.
+  </div> : null;
+};
+
+
+const BuildSummaryTimestamp: React.SFC<BuildSummaryTimestampProps> = ({timestamp}) => <span className="build-pipeline__timestamp text-muted">
+  { fromNow(timestamp) }
+</span>;
+
+const BuildPipelineSummary: React.SFC<BuildPipelineSummaryProps> = ({obj}) => {
+  const buildNumber = getBuildNumber(obj);
+  return <div className="build-pipeline__summary">
+    <div className="build-pipeline__phase">
+      <BuildSummaryStatusIcon status={obj.status.phase} /> Build {buildNumber}
+    </div>
+    <BuildSummaryTimestamp timestamp={obj.metadata.creationTimestamp} />
+    <BuildLogLink obj={obj} />
+  </div>;
+};
+
+const BuildAnimation:React.SFC<BuildAnimationProps> = ({status}) => <div className={`build-pipeline__status-bar build-pipeline__status-bar--${_.kebabCase(status)}`}>
+  <div className="build-pipeline__animation-line"></div>
+  <div className="build-pipeline__animation-circle">
+    <div className="build-pipeline__circle-clip1"></div>
+    <div className="build-pipeline__circle-clip2"></div>
+    <div className="build-pipeline__circle-inner">
+      <div className="build-pipeline__circle-inner-fill"></div>
+    </div>
+  </div>
+</div>;
+
+const JenkinsInputUrl: React.SFC<JenkinsInputUrlProps> = ({obj, stage}) => {
+  const pending = stage && (stage.status === 'PAUSED_PENDING_INPUT');
```

Comment:
I don't think you need to check if stage is defined here. We don't check other places, so it will be an error before it gets to this point.

Reply:
Makes sense. hopefully the check in `build.tsx` means we won't start rendering any of this unless there are jenkins pipeline stages.